sql/sql_table.cc

```diff
@@ -3374,7 +3374,8 @@ bool mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
 
       if (thd->variables.binlog_ddl_skip_rewrite ||
           thd->system_thread == SYSTEM_THREAD_SLAVE_SQL ||
-          thd->system_thread == SYSTEM_THREAD_SLAVE_WORKER) {
+          thd->system_thread == SYSTEM_THREAD_SLAVE_WORKER ||
+          thd->is_binlog_applier()) {
```

Comment:
The rationale is that the query rewrite should be avoided for replication applier and binlog applier threads (during PITR), so that the query is logged as is. So, this should independent of binlog_ddl_skip_rewrite option.

Reply:
:+1: